File: app/android/app/src/main/kotlin/com/pk/ai_keyboard/ai/AiFailure.kt
package com.pk.ai_keyboard.ai

sealed class AiFailure {
    object MissingApiKey : AiFailure()
    object InvalidApiKey : AiFailure()
    object NetworkError : AiFailure()
    object Timeout : AiFailure()
    data class HttpError(val statusCode: Int, val message: String) : AiFailure()
    object InvalidResponse : AiFailure()
    object ContextChanged : AiFailure()
}

File: app/ios/KeyboardExtension/AI/AiFailure.swift
import Foundation

enum AiFailure: Error {
    case missingApiKey
    case invalidApiKey
    case unauthorized
    case forbidden
    case modelNotFound
    case rateLimited
    case timeout
    case network
    case server
    case invalidResponse
    case emptyResponse
    case cancelled
    case contextChanged
    case textTooLong
    case unknown(String)

    var userMessage: String {
        switch self {
        case .missingApiKey: return "⚠️ API key missing"
        case .invalidApiKey, .unauthorized: return "⚠️ Invalid API Key"
        case .forbidden: return "⚠️ Access Denied"
        case .modelNotFound: return "⚠️ Model Not Found"
        case .rateLimited: return "⚠️ Rate Limited"
        case .timeout: return "⚠️ Request Timed Out"
        case .network: return "⚠️ Network Error"
        case .server: return "⚠️ Provider Error"
        case .invalidResponse, .emptyResponse: return "⚠️ Empty AI Response"
        case .cancelled, .contextChanged: return "✨ AtFix"
        case .textTooLong: return "⚠️ Text Too Long"
        case .unknown(let msg): return "⚠️ Error: \(msg)"
        }
    }
}

